Brokerages/InteractiveBrokers/InteractiveBrokersSymbolMapper.cs

```diff
@@ -199,6 +201,56 @@ public string GetLeanRootSymbol(string brokerageRootSymbol)
             return _ibNameMap.ContainsKey(brokerageRootSymbol) ? _ibNameMap[brokerageRootSymbol] : brokerageRootSymbol;
         }
 
+        /// <summary>
+        /// Parses a contract for future with malformed data.
+        /// Malformed data usually manifests itself by having "0" assigned to some values
+        /// we expect, like the contract's expiry date. The contract is returned by IB
+        /// like this, usually due to a high amount of data subscriptions that are active
+        /// in an account, surpassing IB's imposed limit. Read more about this here: https://interactivebrokers.github.io/tws-api/rtd_fqa_errors.html#rtd_common_errors_maxmktdata
+        ///
+        /// We are provided a string in the Symbol in malformed contracts that can be
+        /// parsed to construct the clean contract, which is done by this method.
+        /// </summary>
+        /// <param name="malformedContract">Malformed contract (for futures), i.e. a contract with invalid values ("0") in some of its fields</param>
+        /// <param name="symbolPropertiesDatabase">The symbol properties database to use</param>
+        /// <returns>Clean Contract for the future</returns>
+        /// <remarks>
+        /// The malformed contract returns data similar to the following when calling <see cref="InteractiveBrokersBrokerage.GetContractDetails"/>: ES       MAR2021
+        /// </remarks>
+        public Contract ParseMalformedContractFutureSymbol(Contract malformedContract, SymbolPropertiesDatabase symbolPropertiesDatabase)
+        {
+            Log.Trace($"InteractiveBrokersSymbolMapper.ParseMalformedContractFutureSymbol(): Parsing malformed contract: {InteractiveBrokersBrokerage.GetContractDescription(malformedContract)} with trading class: \"{malformedContract.TradingClass}\"");
+
+            var matches = Regex.Matches(malformedContract.Symbol, @"^(\S*)\s*(\D*)(\d*)");
+
+            var match = matches[0].Groups;
+            var contractSymbol = match[1].Value;
+            var contractMonthExpiration = DateTime.ParseExact(match[2].Value, "MMM", CultureInfo.CurrentCulture).Month;
+            var contractYearExpiration = match[3].Value;
+
+            var leanSymbol = GetLeanRootSymbol(contractSymbol);
+            string market;
+            if (!symbolPropertiesDatabase.TryGetMarket(leanSymbol, SecurityType.Future, out market))
+            {
+                market = InteractiveBrokersBrokerageModel.DefaultMarketMap[SecurityType.Future];
+            }
+            var canonicalSymbol = Symbol.CreateFuture(leanSymbol, market, SecurityIdentifier.DefaultDate);
+            var expirationMonthYear = new DateTime(int.Parse(contractYearExpiration, CultureInfo.InvariantCulture), contractMonthExpiration, 1);
```

Comment:
Nit: I recommend renaming this to something along the lines of `contractMonthYear` or similar, since IB returns the contract month rather than the expiry month of the Future.